salt/_modules/caasp_net.py

```diff
@@ -23,10 +24,13 @@ def get_hosts():
     :return: Dict(ip_address -> [aliases])
     """
     hosts = __salt__['hosts.list_hosts']()  # type: dict
+    # Python3 (correctly) prevents modifying a dictionary we are iterating
+    # over, so we have to copy it first, so we can remove comment entries.
+    hosts_copy = hosts.copy()
     for key in hosts.keys():
         if key.startswith("comment-"):
-            hosts.pop(key)
-    return hosts
+            hosts_copy.pop(key)
+    return hosts_copy
```

Comment:
Taking into account we are using Pyrthon 3, I think it would be much better to use a dictionary comprehension:

```suggestion
    return {k:v for k,v in hosts.items() if not k.startswith("comment-")}
```

Reply:
I will change to this approach - especially since I don't intend to backport these changes, as they are not needed for `python2`.